packages/xstate-fsm/src/index.ts

```diff
@@ -186,8 +182,18 @@ export function createMachine<
 
           const isTargetless = target === undefined;
 
+          const nextStateValue = target ?? value;
+          const nextStateConfig = fsmConfig.states[nextStateValue];
+
+          if (!IS_PRODUCTION && !nextStateConfig) {
+            throw new Error(
+              `State '${nextStateValue}' not found on machine ${
+                fsmConfig.id ?? ''
+              }`
+            );
```

Comment:
Can we DRY this up to reduce code size, since this is the same as the other error?

Reply:
just a note: the code size won't really be reduced since this is a dev-only branch of code so overall DRYing might decrease the readability a little bit for kinda no gains, I'm ok with either way though

Reply:
I wanted to at first, but I realized you would have to resolve the `stateConfig` and the `nextStateConfig` at the same time to throw the same error. I don't see how this is possible with the way the code is written are right now. 

I could also write a function that returns the error message string with the value passed in to avoid re-writing the same line twice, but I don't think that's what you had in mind, correct?

Reply:
> just a note: the code size won't really be reduced since this is a dev-only branch of code so overall DRYing might decrease the readability a little bit for kinda no gains, I'm ok with either way though

Ah, missed that. No worries then!